mllib/src/main/scala/org/apache/spark/ml/feature/Tokenizer.scala

```diff
@@ -39,3 +39,66 @@ class Tokenizer extends UnaryTransformer[String, Seq[String], Tokenizer] {
 
   override protected def outputDataType: DataType = new ArrayType(StringType, false)
 }
+
+
+/**
+ * :: AlphaComponent ::
+ * A regex based tokenizer that extracts tokens using a regex.
+ * Optional additional parameters include enabling lowercase stabdarization, a minimum character
+ * size for tokens as well as an array of stop words to remove from the results.
+ */
+@AlphaComponent
+class RegexTokenizer extends UnaryTransformer[String, Seq[String], RegexTokenizer] {
+
+  val lowerCase = new BooleanParam(this, 
```

Comment:
Add doc with `@group param` to group methods in the generated html doc, e.g., https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/param/sharedParams.scala#L65